stock_picking_back2draft/models/stock.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# © 2016 Lorenzo Battistini - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, api
+from openerp.exceptions import Warning as UserError
+from openerp.tools.translate import _
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    @api.multi
+    def action_back_to_draft(self):
+        state_list = self.mapped('state')
```

Comment:
You can do:

`if any(lambda x: x.state != 'cancel' for x in self.mapped('state')):`